lib/game_box/arena.ex

```diff
@@ -284,6 +289,34 @@ defmodule GameBox.Arena do
     {:noreply, state}
   end
 
+  def handle_cast({:monitor, pid}, state) do
+    Process.monitor(pid)
+
+    pids =
+      state[:pids]
+      |> Enum.concat([pid])
+      |> Enum.uniq()
+
+    {:noreply, Map.put(state, :pids, pids)}
+  end
+
+  @impl true
+  def terminate(:normal, state) do
+    state
+  end
+
+  @impl true
+  def handle_info({:DOWN, _ref, :process, pid, _}, state) do
+    pids = List.delete(state.pids, pid)
+    state = Map.put(state, :pids, pids)
+
+    if Enum.empty?(pids) do
+      {:stop, :normal, state}
+    else
+      {:noreply, state}
+    end
```

Comment:
Can you explain how this works? Does the down message come in any time the process crashes? Here this conditional just tells OTP whether or not to restart it?

Reply:
My understanding is not necessarily any time it crashes but whenever the process that's being monitored exits so could be a crash or could be a normal exit. So this happens when someone navigates away from a page for instance. If there are no remaining pids in the state then we send a stop message and that gets picked up by the terminate which will gracefully let it go away. If the state gets returned with a noreply it will create another one. 

However, I realized this won't quite do it for the arena situation because it will tear down the arena instantly if a host refreshes their page while no one else is also in there.

Reply:
Yeah my feeling was, the way it would work is the arena supervisor would have another child that is just checking all the arenas every x seconds and cleaning them up if it finds one that is empty.

Reply:
I'm not exactly sure how I would do that with how things are set up now but I'll take a look into it

Reply:
I ended up leaving them in the same genservers but adding a bit of a buffer with a send after